lib/facility_factory.rb

```diff
@@ -0,0 +1,28 @@
+class FacilityFactory
+  def create_facilities(data)
+    if data == DmvDataService.new.co_dmv_office_locations
+      data.map do |facility_data|
+      address = "#{facility_data[:address_li]}, #{facility_data[:address__1]}, #{facility_data[:city]}, #{facility_data[:state]} #{facility_data[:zip]}"
+      facility = Facility.new({name: facility_data[:dmv_office],
+        address: address,
+        phone: facility_data[:phone]})
+      end 
+    elsif data == DmvDataService.new.ny_dmv_office_locations
+      data.map do |facility_data|
+      address = "#{facility_data[:street_address_line_1]}, #{facility_data[:city]}, #{facility_data[:state]}, #{facility_data[:zip_code]}"
+      facility = Facility.new({name: facility_data[:office_name],
+        address: address,
+        phone: facility_data[:public_phone_number]})
+      end 
+    elsif data == DmvDataService.new.mo_dmv_office_locations
+      data.map do |facility_data|
+      address = "#{facility_data[:address1]}, #{facility_data[:city]}, #{facility_data[:state]}, #{facility_data[:zipcode]}"
+      facility = Facility.new({name: facility_data[:name],
+        address: address,
+        phone: facility_data[:phone]})
+      end 
+    end 
+  end
```

Comment:
Great question about how to refactor this-- it works well 🎉 but got pretty long and difficult to read.
1. If the return value of `DmvDataService.new.co_dmv_office_locations` for example, is passed to this method as an argument, we shouldn't call that method again if we can avoid it. Especially since this method calls the 'outside world' to get data from an external API, which can be an expensive process (meaning long, but also, you will eventually work with APIs that cost money to ping). 
I would look for some value in the nested dataset which could identify which state it's coming from, and use that in the conditional instead.
2. As far as refactoring, I might start with a helper method for each state which could be called in this method's `maps`:
```
def create_facilities(data)
  if data[:some_value][:some_unique_co_value] # pardon my pseudocode 
    data.map { |facility_data| new_co_facility(facility_data) }
  elsif data[:some_unique_ny_value]
    data.map { |facility_data| new_ny_facility(facility_data) }
  else
    data.map { |facility_data| new_mo_facility(facility_data) }
  end
end
```